supporting_scripts/course-scripts/quick-course-setup/testFiles-template/randomized/RandomizedTestCases.java

```diff
@@ -0,0 +1,131 @@
+package ${packageName};
+
+import org.junit.jupiter.api.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.lang.reflect.InvocationTargetException;
+import java.text.*;
+import java.util.*;
+
+import static de.tum.in.test.api.util.ReflectionTestUtils.*;
+
+import de.tum.in.test.api.BlacklistPath;
+import de.tum.in.test.api.PathType;
+import de.tum.in.test.api.StrictTimeout;
+import de.tum.in.test.api.WhitelistPath;
+import de.tum.in.test.api.jupiter.Public;
+@Public
+@WhitelistPath("target") // mainly for Artemis
+@BlacklistPath("target/test-classes") // prevent access to test-related classes and resources
+class RandomizedTestCases {
+
+    @Test
+    @StrictTimeout(1)
+    void testCase1() {
+        Random random = new Random();
+        boolean shouldFail = random.nextBoolean();
+
+        if (shouldFail) {
+            fail(String.format("Different error: %s, %s", System.currentTimeMillis(), random.nextInt(Integer.MAX_VALUE)));
+        }
+    }
```

Comment:
That's like 10 times the same code (only different method names), I guess it would make sense to put the test content into one small method, which is called by each test case. That would get rid of a lot of duplicated code. Or am I missing something here?

Reply:
yes your right, thanks 👍